File: python/src/factorial/factorial.py
# recursive formula: n x m!
# where m = n - 1
def factorial(n):
    try:
        if n > 1:
            m = n - 1

            return n * factorial(m)
        elif n == 1:
            return 1
        else:
            print("You can't enter negative values.")
    except TypeError:
        print("Please enter a number.")

File: javascript/src/factorial/factorial.ts
export const factorial = (n: number): number => {
  if (n > 1) {
    const m = n - 1;

    return n * factorial(m);
  } else if (n === 1) {
    return 1;
  } else {
    console.log("You can't enter negative values.");
    return 0;
  }
};

